src/tests/multihost/alltests/conftest.py

```diff
@@ -1062,6 +1062,50 @@ def remove_journalsssd():
     request.addfinalizer(remove_journalsssd)
 
 
+@pytest.fixture(scope='class')
+def sssdproxyldap_test(session_multihost, request):
+    """ Backup and restore sssd.conf """
+    tools = sssdTools(session_multihost.client[0])
+    execute_cmd(session_multihost, "> /etc/pam.d/sssdproxyldap")
+    execute_cmd(session_multihost, "echo 'auth        required      "
+                                   "pam_ldap.so' > /etc/pam.d/sssdproxyldap")
+    execute_cmd(session_multihost, "echo 'account     required      "
+                                   "pam_ldap.so' >> /etc/pam.d/sssdproxyldap")
+    execute_cmd(session_multihost, "echo 'password    required      "
+                                   "pam_ldap.so' >> /etc/pam.d/sssdproxyldap")
+    execute_cmd(session_multihost, "echo 'session     required      "
+                                   "pam_ldap.so' >> /etc/pam.d/sssdproxyldap")
+    domain_name = tools.get_domain_section_name()
+    domain_params = {'proxy_pam_target': 'sssdproxyldap',
+                     'id_provider': 'proxy',
+                     'proxy_lib_name': 'ldap'}
+    tools.sssd_conf('domain/' + domain_name, domain_params)
+    execute_cmd(session_multihost, "> /etc/nslcd.conf")
+    execute_cmd(session_multihost, "echo 'uid nslcd' > /etc/nslcd.conf")
+    execute_cmd(session_multihost, "echo 'gid ldap' >> /etc/nslcd.conf")
+    execute_cmd(session_multihost, f"echo "
+                                   f"'uri ldap://{session_multihost.master[0].ip}' "
+                                   f">> /etc/nslcd.conf")
+    execute_cmd(session_multihost, f"echo 'base {ds_suffix}' "
+                                   f">> /etc/nslcd.conf")
+    execute_cmd(session_multihost, "systemctl restart nslcd")
+    file_location = '/sssd/src/tests/multihost/alltests/script/sssdproxyldap.sh'
+    session_multihost.client[0].transport.put_file(os.getcwd()
+                                                   + file_location,
+                                                   '/tmp/sssdproxyldap.sh')
+    execute_cmd(session_multihost, f"chmod 755 /tmp/sssdproxyldap.sh")
+    session_multihost.master[0].run_command("kadmin.local -q "
+                                            "'addprinc -pw "
+                                            "Secret123 foo2@example1'")
+    tools.clear_sssd_cache()
+
+    def restore_sssdproxyldap_test():
+        """ Restore"""
+        session_multihost.client[0].run_command("rm -vf "
+                                                "/tmp/sssdproxyldap.sh")
```

Comment:
Cleanup of nslcd.conf and /etc/pam.d/sssdproxyldap. Do we also need to delete the krb principal?

Reply:
Cleanup of nslcd.conf and /etc/pam.d/sssdproxyldap is done , no need to delete the krb principal Secret123 is default password for all exiting ldap users